src/village/mod.rs

```diff
@@ -23,39 +25,33 @@ impl Village {
         village
     }
 
-    pub fn add_resource(&mut self, resource_type: ResourceType) {
-        for resource in self.resources.iter() {
-            if resource.resource_type == resource_type {
-                return;
-            }
-        }
-
+    pub fn create_resource(&mut self, resource_type: ResourceType) {
         self.resources.push(Resource {
             resource_type: resource_type,
-            worker_count: 0
+            worker_count: 0,
+            resource_id: self.number_of_resource_sites,
         });
+        self.number_of_resource_sites += 1;
+    }
+    
+    pub fn assign_workers(&mut self, resource_id: u32, number_of_workers: u32) {
+        if self.idle_worker_count() >= number_of_workers && resource_id < self.number_of_resource_sites {
+            self.resources[resource_id as usize].worker_count += number_of_workers;
```

Comment:
resource_id does not always map to position in the resources vector (e.g. after an element is removed)